src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs

```diff
@@ -49,14 +49,22 @@ public CrpgAgentApplyDamageModel(CrpgConstants constants)
         if (collisionData.AttackBlockedWithShield && finalDamage > 0)
         {
             int shieldSkill = GetSkillValue(attackInformation.VictimAgentOrigin, CrpgSkills.Shield);
-            finalDamage /= MathHelper.RecursivePolynomialFunctionOfDegree2(shieldSkill, _constants.DurabilityFactorForShieldRecursiveCoefs) * 1.25f;
-            if (weapon.CurrentUsageItem.WeaponFlags.HasAnyFlag(WeaponFlags.BonusAgainstShield) && meleeClass.Contains(weapon.CurrentUsageItem.WeaponClass))
+            finalDamage /= MathHelper.RecursivePolynomialFunctionOfDegree2(shieldSkill, _constants.DurabilityFactorForShieldRecursiveCoefs);
+            if (meleeClass.Contains(weapon.CurrentUsageItem.WeaponClass))
             {
-                // this bonus is on top of the native x2 in MissionCombatMechanicsHelper
-                // so the final bonus is 3.5. We do this instead of nerfing the impact of shield skill so shield can stay virtually unbreakable against sword.
-                // it is the same logic as arrows not dealing a lot of damage to horse but spears dealing extra damage to horses
-                // As we want archer to fear cavs and cavs to fear spears, we want swords to fear shielders and shielders to fear axes.
-                finalDamage *= 2f;
+                finalDamage *=
+                    (collisionData.StrikeType == (int)StrikeType.Thrust)
+                        ? weapon.CurrentUsageItem.ThrustDamageFactor
+                        : weapon.CurrentUsageItem.SwingDamageFactor;
```

Comment:
Coud you comment how you find out about that part. It doesn't seem intuitive.

Also I think you should get a warning for the redundant paranthesis.